smb/smb.go

```diff
@@ -1572,3 +2256,39 @@ func (s *Session) NewSetInfoReq(share string, fileId []byte) (SetInfoReq, error)
 		FileId:        fileId,
 	}, nil
 }
+
+func (s *Session) NewQueryInfoReq(
+	share string,
+	fileId []byte,
+	infoType byte,
+	fileInformationClass byte,
+	additionalInformation uint32,
+	flags uint32,
+	outputBufferLength uint32,
+	inputBuffer []byte,
+) (QueryInfoReq, error) {
+	header := newHeader()
+	header.Command = CommandQueryInfo
+	header.CreditCharge = 1
```

Comment:
Replace line with
```
header.CreditCharge = calcCreditCharge(outputBufferLength)
```
To properly handle larger values of outputBufferLength where more than 1 credit is required.